### ISSUE TITLE
`-dir` flag not being respected

### ISSUE THREAD

**Issue by berwyn** [angular/angular-cli#5867]
### Bug Report or Feature Request (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
```

### Versions.

@angular/cli: 1.0.0
node: 7.7.1
os: win32 x64 (Windows 10 1703 Build 15063.14 "Creators Update" RC)
@angular/common: 4.0.1
@angular/compiler: 4.0.1
@angular/core: 4.0.1
@angular/flex-layout: 2.0.0-beta.7
@angular/http: 4.0.1
@angular/material: 2.0.0-beta.2
@angular/platform-browser: 4.0.1
@angular/platform-browser-dynamic: 4.0.1

### Repro steps.

1. run `ng new -dir foo myapp`
2. Notice how the app is created in cwd, not in `./foo`

### The log given by the failure.

Correct output for app generation

### Desired functionality.

The app should be created in `./foo`, as indicated in the help text:
```
--directory (String) The directory name to create the app in.
  aliases: -dir <value>, --directory <value>
```

**Comment by delasteve** [comment]
I was unable to reproduce this using 

```
@angular/cli: 1.0.1
node: 7.9.0
os: win32 x64
```

Using `ng new -dir foo someapp`, when I `ls`ed the parent directory, I got 
```
Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----        4/27/2017  11:58 AM                foo
```

**Comment by berwyn** [comment]
My apologies, I debugged it more and found that the issue required more specific conditions to dupe and forgot to update this bug report. This is specifically if a directory above contains a `package.json` file. In my case, I was building a group of modules with lerna, where one was an Angular app. Angular CLI 1.0.0 would create the app at the top-level next to `package.json` rather than in the directory specified by `-dir`.

In other words, my expected setup was
```bash
ng new -dir packages/foo-app foo
```
resulting in
```
./
|- package.json # Top level
|- lerna.json
`- packages
   `- foo-app
     |- .angular-cli.json
     |- karma.conf.js
     ...
```

but instead I got
```
./
|- package.json
|- lerna.json
|- .angular-cli.json
|- karma.conf.js
...
```

**Comment by filipesilva** [comment]
I actually thought we fixed this already in https://github.com/angular/angular-cli/pull/4901. Maybe something is broken.

**Comment by hansl** [comment]
This has been fixed a long time ago, never marked. If this still happens please create a new issue so we can follow up.
